app/controllers/saml_idp_controller.rb

```diff
@@ -15,6 +16,20 @@ class SamlIdpController < ApplicationController
   before_action :store_sp_data, only: :auth
   before_action :confirm_two_factor_authenticated, except: [:metadata, :logout]
 
+  helper_method :saml_response_url
+
+  def raw_algorithm
+    SamlIdp.config.algorithm
+  end
+
+  def signature_opts
+    algorithm
+  end
```

Comment:
Same comment for the two methods above, which I believe are already defined in the saml_idp gem.